src/xcode/ENA/ENA/Source/Models/Exposure/MockENManager.swift

```diff
@@ -27,15 +25,18 @@ final class MockENManager: NSObject {
 	// MARK: - Activating
 
 	func activate(completionHandler: @escaping ENErrorHandler) {
-		DispatchQueue.main.async {
+		dispatchQueue.async {
 			completionHandler(nil)
 		}
 	}
 
 	func setExposureNotificationEnabled(_ enabled: Bool, completionHandler: @escaping ENErrorHandler) {
-		self.enabled = enabled
-		DispatchQueue.main.async {
-			completionHandler(nil)
+		let dispatchQueue = self.dispatchQueue
```

Comment:
Do the dispatchQueue really has to be reused? And why?

Reply:
Good point. The idea was to use only local variables in the closure ... but now I see that is not possible here ... so maybe a guard that self is still available?
... and, by the way, I see similar issues in my other closures ... ooops ...

Reply:
Yeah i like your approach. Waiting for your fixes :)

Reply:
@pascal-brause please have a look at my latest commit on this branch